client/rest-high-level/src/main/java/org/elasticsearch/client/ml/inference/TrainedModelDefinition.java

```diff
@@ -124,14 +138,69 @@ public Builder setTrainedModel(TrainedModel trainedModel) {
             return this;
         }
 
+        public Builder setInput(Input input) {
+            this.input = input;
+            return this;
+        }
+
         private Builder setTrainedModel(List<TrainedModel> trainedModel) {
             assert trainedModel.size() == 1;
             return setTrainedModel(trainedModel.get(0));
         }
 
         public TrainedModelDefinition build() {
-            return new TrainedModelDefinition(this.trainedModel, this.preProcessors);
+            return new TrainedModelDefinition(this.trainedModel, this.preProcessors, this.input);
+        }
+    }
+
+    public static class Input implements ToXContentObject {
```

Comment:
Should this class have its own serialization unit tests?

Reply:
I don't think so. Its so small and is only used within this definition. Coverage is handled via the TrainedModelDefinitionTests